packages/firestore/src/remote/remote_event.ts

```diff
@@ -242,4 +219,19 @@ export class UpdateMapping {
       this.removedDocuments.isEqual(other.removedDocuments)
     );
   }
+
+  /**
+   * Strips out mapping changes that aren't actually changes. That is, if the document already
+   * existed in the target, and is being added in the target, and this is not a reset, we can
+   * skip doing the work to associate the document with the target because it has already been done.
+   */
+  filterUpdates(existingKeys: DocumentKeySet): void {
```

Comment:
The name of this functions is kind of strange in this context (`UpdateMapping.fitlerUpdates`). I can't think of a much better name right now (and we would probably want to update the name of the class rather than the name of this function). Feel free to leave as is unless you can think of something better.

Reply:
I'm going to leave it as-is to match other clients. If someone were to do some additional work on this method and prefer a different name, I'm sure they could rename it then and I'd be happy to approve it.